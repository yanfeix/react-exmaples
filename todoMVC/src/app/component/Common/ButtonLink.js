import React from 'react';
import PropTypes from 'prop-types';

class ButtonLink extends React.Component {
    constructor(props) {
        super(props);

        this.handleClick = this.handleClick.bind(this);
    }

    handleClick() {
        const { onClick, filterType } = this.props;
        onClick(filterType);
    }

    render() {
        const { text } = this.props;
        return (
            <a href="#" onClick={this.handleClick}>{ text }</a>
        );
    }
}

ButtonLink.propTypes = {
    text: PropTypes.string.isRequired,
    onClick: PropTypes.func.isRequired,
    filterType: PropTypes.string.isRequired,
};

export default ButtonLink;
