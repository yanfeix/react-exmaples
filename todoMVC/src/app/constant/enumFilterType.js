const FilterType = {
    ALL: 'SHOW_ALL',
    ACTIVE: 'SHOW_ACTIVE',
    COMPLETED: 'SHOW_COMPLETED',
};

export default FilterType;
