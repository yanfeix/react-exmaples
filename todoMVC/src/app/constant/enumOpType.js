const OpType = {
    ADD: 'ADD',
    DELETE: 'DELETE',
    EDIT: 'EDIT',
    ACTIVATE: 'ACTIVATE',
    COMPLETE: 'COMPLETE',
    CLEAR_COMPLETED: 'CLEAR COMPLETED',
};

export default OpType;
