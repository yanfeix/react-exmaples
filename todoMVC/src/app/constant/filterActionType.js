const FilterAction = {
    SET_FILTER_TYPE: 'SET_FILTER_TYPE',
};

export default FilterAction;
