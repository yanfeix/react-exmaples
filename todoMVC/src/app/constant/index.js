export { default as OpType } from './enumOpType';
export { default as FilterType } from './enumFilterType';
export { default as UserAction } from './userActionType';
export { default as FilterAction } from './filterActionType';
