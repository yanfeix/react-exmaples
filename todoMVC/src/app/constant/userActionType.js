const UserAction = {
    ADD: 'ADD_TODO',
    REMOVE: 'REMOVE_TODO',
    MODIFY: 'MODIFY_TODO',
    ACTIVATE: 'ACTIVATE_TODO',
    COMPLETE: 'COMPLETE_TODO',
    CLEAR_COMPLETED: 'CLEAR_COMPLETED',
    TOGGLE: 'TOGGLE_ALL',
};

export default UserAction;
