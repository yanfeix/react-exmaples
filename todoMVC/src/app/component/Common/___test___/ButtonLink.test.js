import React from 'react';
import renderer from 'react-test-renderer';
import ButtonLink from '../ButtonLink';
import { FilterType } from '../../../constant';

it(' ButtonLink is created and click event is captured ', () => {
    const mockOnClicked = jest.fn();
    const component = renderer.create(
        <ButtonLink text="Completed" filterType={FilterType.COMPLETED} onClick={mockOnClicked} />,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.onClick();
    expect(mockOnClicked).toHaveBeenCalled();
});
