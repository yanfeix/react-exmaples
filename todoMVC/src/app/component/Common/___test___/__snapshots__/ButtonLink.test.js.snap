// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` ButtonLink is created and click event is captured  1`] = `
<a
  href="#"
  onClick={[Function]}
>
  Completed
</a>
`;
